src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py

```diff
@@ -0,0 +1,139 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+'''
+gridsearch_tuner.py including:
+    class GridSearchTuner
+'''
+
+import copy
+import numpy as np
+
+import nni
+from nni.tuner import Tuner
+
+TYPE = '_type'
+CHOICE = 'choice'
+VALUE = '_value'
+
+
+class GridSearchTuner(Tuner):
+    '''
+    GridSearchTuner will search all the possible configures that the user define in the searchSpace.
+    The only acceptable types of search space are 'quniform', 'qloguniform' and 'choice'
+
+    Type 'choice' will select one of the options. Note that it can also be nested.
+
+    Type 'quniform' will receive three values [low, high, q], where [low, high] specifies a range
+    and 'q' specifies the number of values that will be sampled evenly.
+    It will be sampled in a way that the first sampled value is 'low', and each of the following values is (high-low)/q
+    larger that the value in front of it.
+
+    Type 'qloguniform' behaves like 'quniform' except that it will first change the range to [log10(low), log10(high)]
+    and sample and then change the sampled value back.
+    '''
+
+    def __init__(self, optimize_mode):
+        self.count = -1
+        self.expanded_search_space = []
+
+    def json2paramater(self, ss_spec):
+        '''
+        generate all possible config for hyperparameters from hyperparameter space.
+        ss_spec: hyperparameter space
+        '''
+        if isinstance(ss_spec, dict):
+            if '_type' in ss_spec.keys():
+                _type = ss_spec['_type']
+                _value = ss_spec['_value']
+                chosen_params = list()
+                if _type == 'choice':
+                    for value in _value:
+                        choice = self.json2paramater(value)
+                        if isinstance(choice, list):
+                            chosen_params.extend(choice)
+                        else:
+                            chosen_params.append(choice)
+                else:
+                    chosen_params = self._parse_parameter(_type, _value)
+            else:
+                chosen_params = dict()
+                for key in ss_spec.keys():
+                    chosen_params[key] = self.json2paramater(ss_spec[key])
+                return self.expand_parameters(chosen_params)
+        elif isinstance(ss_spec, list):
+            chosen_params = list()
+            for subspec in ss_spec[1:]:
+                choice = self.json2paramater(subspec)
+                if isinstance(choice, list):
+                    chosen_params.extend(choice)
+                else:
+                    chosen_params.append(choice)
+            chosen_params = list(map(lambda v: {ss_spec[0]: v}, chosen_params))
+        else:
+            chosen_params = copy.deepcopy(ss_spec)
+        return chosen_params
+
+    def _parse(self, param_value):
+        low, high, count = param_value[0], param_value[1], max(2, param_value[2])
```

Comment:
if  the min value of param_value[2] is 2 , we should raise error if  param_value[2] < 2.

Reply:
maybe rename _parse to a meaningful name ?

Reply:
ok, will rename some functions to be more meaningful and raise error